daemon/tests/harness/mocks/monitor.rs

```diff
@@ -40,6 +42,78 @@ impl MonitorActor {
 }
 
 #[derive(Default)]
-pub struct MockMonitor {}
+pub struct MockMonitor {
+    executor: Option<command::Executor>,
+}
+
+impl MockMonitor {
+    pub fn set_executor(&mut self, executor: command::Executor) {
+        self.executor = Some(executor);
+    }
+
+    pub async fn confirm_lock_transaction(&mut self, id: OrderId) {
+        self.executor
+            .as_ref()
+            .expect("executor to be set during test setup")
+            .execute(id, |cfd| Ok(cfd.handle_lock_confirmed()))
```

Comment:
I find it a bit weird that we have to mock all of these. I wonder if we could come up with a design where we can use production code to achieve this, i.e. arrange the production code in a way that we trigger the executor, but don't use the state management in the production monitor.

Reply:
Very interesting idea. We would have to make parts of the original monitor actor swappable.

The internal logic is already split into the IO component, updating the state sending out events. We would have to plug the mock in before the last bit.

Interesting to explore but also, not that important I feel? Esp. keeping monitor::Event private is something I'd like to keep.